src/pipes_filters/Input.java

```diff
@@ -1,5 +1,24 @@
 package pipes_filters;
+import java.io.FileInputStream;
+import java.io.IOException;
 
-public class Input {
-
+public class Input extends Filter {
+	FileInputStream fileIn;
+	
+	public Input(FileInputStream in, Pipe entry, Pipe out) {
```

Comment:
It seems that input is special filter. It reads data from the inputStream rather than a pipe. In the constructor. What should we pass as the entry? Or we could create another constructor in the Filter that receives two parameters. One is FileInputStream in and another is Pipe out